### PR TITLE
Update ArraySource.php

### PR DESCRIPTION
**Pull request by 512banque**
Fixing the problem of Paginator sorting not working.
Solving this problem: https://github.com/cakephp/datasources/issues/95

### CODE REVIEW

File: Model/Datasource/ArraySource.php [hunk 1]
```diff
@@ -164,6 +164,10 @@ public function read(Model $model, $queryData = array(), $recursive = null) {
 			}
 			return array(array(array('count' => count($data))));
 		}
+		// Sort and Direction must overwrite Order
+		if (!empty($queryData['sort'])) {
+			$queryData['order'] = array($queryData['sort'].' '.((empty($queryData['direction'])) ? 'ASC' : strtoupper($queryData['direction'])));
```
Comment (jadb) [inline Model/Datasource/ArraySource.php]: Breaking this into more than one line would be more readable no?

**Comment by jadb** [comment]
Thanks. Can you add test case(s) for it?

**Comment by jrbasso** [comment]
It is weird to have the datasource class reading the `sort` field. It is something that should be done by the Paginator/Model. I don't recall of DboDatasource checking this field. It is not part of the standard communication between Model and Datasources. Or am I wrong?

**Comment by jadb** [comment]
@jrbasso you could very well be right. 

I asked for the test to see how it is being used. I didn't dig more into it before I could see a test case as I don't use that part myself.

**Comment by chinpei215** [comment]
Closing, as I opened an alternative pull request #120.

**Pull request closed without merging**